Libraries/Modal/Modal.js

```diff
@@ -138,6 +153,18 @@ class Modal extends React.Component {
   static contextTypes = {
     rootTag: PropTypes.number,
   };
+  constructor(props: Object) {
+    super(props);
+    Modal._confirmProps(props);
+  }
+  componentWillReceiveProps(nextProps: Object) {
+    Modal._confirmProps(nextProps);
+  }
+  static _confirmProps(props: Object) {
+    if (props.presentationStyle && props.presentationStyle !== 'overFullScreen' && props.transparent) {
+      console.warn(`Modal with '${props.presentationStyle}' presentation style and 'transparent' value is not supported.`);
```

Comment:
@shergin I couldn't find where in the Swift/Obj-C documentation it says using presentationStyle with transparent it's invalid.

https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/formsheet